ksql-engine/src/main/java/io/confluent/ksql/metrics/ConsumerCollector.java

```diff
@@ -0,0 +1,130 @@
+ * limitations under the License.
+ **/
+
+package io.confluent.ksql.metrics;
+
+import io.confluent.common.metrics.KafkaMetric;
+import io.confluent.common.metrics.MetricName;
+import io.confluent.common.metrics.Metrics;
+import io.confluent.common.metrics.Sensor;
+import io.confluent.common.metrics.stats.Rate;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+public class ConsumerCollector {
+
+  final Map<String, Counter> topicCounters = new HashMap<>();
+  private Metrics metrics;
+  private String id;
+
+  public ConsumerCollector(Metrics metrics) {
+    this.metrics = metrics;
+  }
+
+
+  public ConsumerRecords onConsume(ConsumerRecords records) {
+    if (id == null) {
+      throw new RuntimeException(new IllegalStateException("Cannot run without 'id' being set, use ConsumerCollector.configure(map)"));
+    }
+
+    collect(records);
+    return records;
+  }
+
+  private void collect(ConsumerRecords consumerRecords) {
+    Stream<ConsumerRecord> stream = StreamSupport.stream(consumerRecords.spliterator(), false);
+    stream.forEach((record) -> {
+      topicCounters.computeIfAbsent(record.topic(), k -> new Counter(id, metrics)).increment(record);
+    });
+  }
+  public void configure(Map<String, ?> map) {
+    this.id = (String) map.get(ConsumerConfig.GROUP_ID_CONFIG);
+  }
+
+  public void close() {
+  }
+
+  public String statsForTopic(String topic) {
+    return topicCounters.containsKey(topic) ? "Consumer:" + topicCounters.get(topic).toString() : "";
+  }
+
+  public static class Counter {
+    private final String id;
+    private final Metrics metrics;
+    private Sensor sensor;
+    private KafkaMetric rate;
+
+    public Counter(String id, Metrics metrics) {
+      this.id = id;
+      this.metrics = metrics;
+    }
+
+    Map<Integer, PartitionCounter> partitionCounter = new HashMap<>();
+    long records;
+    long valueBytes;
+
+    public void increment(ConsumerRecord record) {
+      records +=1;
+      valueBytes += record.serializedValueSize();
+
+      if (sensor == null) {
+        String name = "consumer-" + id + "-" +record.topic() + "-" + record.partition();
+        sensor = metrics.sensor(name);
+        System.out.println("Registering:" + name);
+        new RuntimeException("Register:" + name).printStackTrace();
+        MetricName producerRate = new MetricName(name + "-rate-per-sec", name, "consumer-stats");
+        sensor.add(producerRate, new Rate(TimeUnit.SECONDS));
+        rate = metrics.metrics().get(producerRate);
+      }
+      sensor.record(1);
+
+      partitionCounter.computeIfAbsent(record.partition(), k -> new PartitionCounter()).increment(record);
```

Comment:
Do we not want to record a rate per partition to? So should this also have another `Sensor`?